core/src/setup/Kokkos_Setup_Cuda.hpp

```diff
@@ -56,6 +56,10 @@
 #define KOKKOS_LAMBDA [=] __host__ __device__
 #define KOKKOS_CLASS_LAMBDA [ =, *this ] __host__ __device__
 
+#if defined(__clang__)
+#define KOKKOS_DEDUCTION_GUIDE __host__ __device__
+#endif
```

Comment:
I'd prefer if we removed the guard. `nvcc` seems to be fine with it as well.

Reply:
I am fine with both solutions though this introduces a change for a working compiler.